sdk/python/lib/pulumi/resource.py

```diff
@@ -894,7 +895,13 @@ def __init__(
         # provided properties and options assigned to this resource.
         parent = opts.parent
```

Comment:
Nit: this can be simplified with `or`:
```suggestion
        parent = opts.parent or ambient_parent.get() or get_root_resource()
```